lib/messages.js

```diff
@@ -713,6 +719,40 @@ const EncryptionDescriptor = {
   }
 }
 
+const EncryptionInfo = {
+  preencode (state, e) {
+    c.uint.preencode(state, e.legacyBlocks)
+  },
+  encode (state, e) {
+    c.uint.encode(state, e.legacyBlocks)
+  },
+  decode (state) {
+    return {
+      legacyBlocks: c.uint.decode(state)
+    }
+  }
+}
+
+const ManifestData = {
+  preencode (state, d) {
+    c.uint.preencode(state, 0)
+    EncryptionInfo.preencode(state, d.encryption)
```

Comment:
easier down the line if we inline this top level instead of nesting (just easier to port to hyperschema)